tests/integration/utils.ts

```diff
@@ -26,30 +26,50 @@ import type {
 } from '@kiltprotocol/types'
 import { Crypto } from '@kiltprotocol/utils'
 
-import { makeSigningKeyTool } from '../testUtils/index.js'
+import { makeSigningKeyTool } from '../testUtils/TestUtils.js'
 
 export const EXISTENTIAL_DEPOSIT = new BN(10 ** 13)
 const ENDOWMENT = EXISTENTIAL_DEPOSIT.muln(10000)
 
 const WS_PORT = 9944
 
-async function getStartedTestContainer(): Promise<StartedTestContainer> {
-  try {
-    const image =
-      process.env.TESTCONTAINERS_NODE_IMG || 'kiltprotocol/mashnet-node'
-    console.log(`using testcontainer with image ${image}`)
-    const testcontainer = new GenericContainer(image)
-      .withCommand(['--dev', `--ws-port=${WS_PORT}`, '--ws-external'])
-      .withExposedPorts(WS_PORT)
-      .withWaitStrategy(Wait.forLogMessage(`:${WS_PORT}`))
-    const started = await testcontainer.start()
-    return started
-  } catch (error) {
-    console.error(
-      'Could not start the docker container via testcontainers, run with DEBUG=testcontainers* to debug'
-    )
-    throw error
+export async function getStartedTestContainer(
+  hostPort?: number
+): Promise<StartedTestContainer> {
+  const image =
+    process.env.TESTCONTAINERS_NODE_IMG || 'kiltprotocol/mashnet-node'
+  console.log(`using testcontainer with image ${image}`)
+  const strategies = [
+    ['--dev', '--ws-external', `--ws-port=${WS_PORT}`],
+    ['--dev', '--rpc-external', `--rpc-port=${WS_PORT}`],
+  ]
+  // eslint-disable-next-line no-plusplus
+  for (let strategy = 0; strategy < strategies.length; strategy++) {
+    console.log(`attempting to launch container using strategy ${strategy}`)
+    try {
```

Comment:
I think this should be possible: 

```suggestion
  for (const strategy of strategies) {
    console.log(`Attempting to launch container using strategy ${strategy}`);
    try {
      const testcontainer = new GenericContainer(image)
        .withCommand(strategy)
        .withExposedPorts(
          typeof hostPort === 'number'
            ? { host: hostPort, container: WS_PORT }
            : WS_PORT
        )
        .withWaitStrategy(Wait.forLogMessage(`:${WS_PORT}`));

      const started = await testcontainer.start();
      console.log('Container started and ready');
      return started;
    } catch (error) {
      console.warn('Failed to start container due to the following error:\n', error);
    }
  }
```